planet/src/main/webapp/places.js

```diff
@@ -148,62 +158,61 @@ function addPlaceDetails(places) {
-        let placeDetails = {Name: '', Rating: '', Address: '', Photo: '',Phone: '',Hours: '',Website: '',Opening: '',Closing: ''}; 
-        
-        // Fetch url and add photo reference to a list
-        promises = [];
-        promises.push(fetch(url).then(response => response.json()).then(function(response) {
-            placeDetails['Name'] = place.name; 
-            // Check if place details exist
-            if (place.rating != undefined) {
-                placeDetails['Rating'] = place.rating;    
-            }
-            if (response.result.formatted_address != undefined) {
-                placeDetails['Address'] = response.result.formatted_address;
-            }
-            if (response.result.photos != undefined) {
-                placeDetails['Photo'] = response.result.photos[0].photo_reference;
-            }
-            if (response.result.formatted_phone_number != undefined) {
-                placeDetails['Phone'] = response.result.formatted_phone_number;
-            }
-            if (response.result.opening_hours != undefined){
-                placeDetails['Hours'] = response.result.opening_hours.weekday_text;
-                // Open and closing time for Monday (MVP purposes)
-                if (response.result.opening_hours.periods[1] != undefined) {
-                    placeDetails['Opening'] = response.result.opening_hours.periods[1].open.time;
-                    if (response.result.opening_hours.periods[1].close != undefined) {
-                        placeDetails['Closing'] = response.result.opening_hours.periods[1].close.time;
-                    }
-                }
-            }
-            if (response.result.website != undefined) {
-                placeDetails['Website'] = response.result.website;
-            }
-
-           
-            placeInfo.push(placeDetails);
-        }));
+        // Make request with fields
+        var place = {
+            placeId: place.place_id,
+            fields: ['name','rating','formatted_phone_number','formatted_address',
+            'opening_hours','photos','url']
+        };
+        // Call Places Details Request
+        let service = new google.maps.places.PlacesService(map);
+        service.getDetails(place, callback); 
     });
-    // Wait for promises to complete before proceeding
-    Promise.all(promises).then(function() {
-        listResults();
-    })
 }
 
-/** Get URL for Place Photo Request  */
-function getPhotoURL(photo_reference) {
-    let baseURL = 'https://maps.googleapis.com/maps/api/place/photo?';
-    let maxWidth = '400';
-    let maxHeight = '200';
-    let photoURL = baseURL + 'maxwidth=' + maxWidth + '&maxheight=' + maxHeight 
-    + '&photoreference=' + photo_reference + '&key=AIzaSyDK36gDoYgOj4AlbCqh1IuaUuTlcpKF0ns'; 
+/** Callback function to handle place details response */
+function callback(place, status) {
```

Comment:
Is it possible to give this function a more descriptive name?

Reply:
This conversation was resolved but it doesn't look like the comment was addressed. Is there a missing commit that needs to be included in the PR?